zetta_utils/mazepa/autoexecute_task_queue.py

```diff
@@ -22,36 +26,58 @@ class AutoexecuteTaskQueue(MessageQueue):
     tasks_todo: list[Task] = attrs.field(init=False, factory=list)
     debug: bool = False
     handle_exceptions: bool = False
+    parallel_if_pool_exists: bool = False
 
     def push(self, payloads: Iterable[Task]):
         # TODO: Fix progress bar issue with multiple live displays in rich
         # for task in track(tasks, description="Local task execution..."):
         self.tasks_todo += list(payloads)
 
     def pull(
-        self, max_num: int = 1, max_time_sec: float = 0  # pylint: disable=unused-argument
+        self,
+        max_num: int = 128,
+        max_time_sec: float = 0,  # pylint: disable=unused-argument
     ) -> list[ReceivedMessage[OutcomeReport]]:
         if max_time_sec != 0:
             raise NotImplementedError()
+        if len(self.tasks_todo) == 0:
+            return []
+        else:
+            pool = get_persistent_process_pool()
+            if not self.parallel_if_pool_exists or pool is None:
+                results: list[ReceivedMessage[OutcomeReport]] = []
+                for task in self.tasks_todo[:max_num]:
+                    results.append(execute_task(task, self.debug, self.handle_exceptions))
+            else:  # pragma: no cover
+                # cannot test within pytest due to picklability
+                futures = []
+                for task in self.tasks_todo[:max_num]:
+                    futures.append(
+                        pool.schedule(execute_task, (task, self.debug, self.handle_exceptions))
```

Comment:
nitpick: would it be better to use `kwargs` here?

[https://pebble.readthedocs.io/en/latest/#pebble-module]